### PR TITLE
Feature/account fiscal position rule migration 8.0

### PR DESCRIPTION
**Pull request by renatonlima**

### CODE REVIEW

File: account_fiscal_position_rule/account_fiscal_position_rule.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+            addrs['invoice'] = self.env['res.partner'].browse(
+                partner_invoice_id)
+
+        # In picking case the invoice_id can be empty but we need a
+        # value I only see this case, maybe we can move this code in
+        # fiscal_stock_rule
+        else:
+            partner_addr = partner.address_get(['invoice'])
+            addr_id = partner_addr['invoice'] and \
+                partner_addr['invoice'] or None
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule.py]: Please do not use `... and ... or ...` but `... if ... else ...`, and do not use `\` but parenthesis.

File: account_fiscal_position_rule/account_fiscal_position_rule.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        result['value'].update(self.fiscal_position_map(**kwargs))
+        return result
+
+
+class AccountFiscalPositionRuleTemplate(models.Model):
+    _name = 'account.fiscal.position.rule.template'
+    _description = 'Account Fiscal Position Rule Template'
+    _order = 'sequence'
+
+    name = fields.Char('Name', size=64, required=True)
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule.py]: `size` to remove: it should be set only when there is a reason to limit (such as an ean13 for instance)

File: account_fiscal_position_rule/account_fiscal_position_rule.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    description = fields.Char('Description', size=128)
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule.py]: `size` to remove: it should be set only when there is a reason to limit (such as an ean13 for instance)

File: account_fiscal_position_rule/account_fiscal_position_rule.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+
+from openerp import models, fields, api
+
+
+class AccountFiscalPositionRule(models.Model):
+    _name = 'account.fiscal.position.rule'
+    _description = 'Account Fiscal Position Rule Template'
+    _order = 'sequence'
+
+    name = fields.Char('Name', size=64, required=True)
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule.py]: `size` to remove: it should be set only when there is a reason to limit (such as an ean13 for instance)

File: account_fiscal_position_rule/account_fiscal_position_rule.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    description = fields.Char('Description', size=128)
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule.py]: `size` to remove: it should be set only when there is a reason to limit (such as an ean13 for instance)

File: account_fiscal_position_rule/account_fiscal_position_rule_view.xml [hunk 6]
```diff
@@ -0,0 +1,223 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+
+	<data>
```
Comment (guewen) [inline account_fiscal_position_rule/account_fiscal_position_rule_view.xml]: Needs to be reindented in order to use 2 spaces instead of tabs.

**Comment by guewen** [comment]
This module deserves tests.

**Comment by guewen** [comment]
Thanks for the work on the migration

File: account_fiscal_position_rule/account_invoice.py [hunk 7]
```diff
@@ -20,45 +20,44 @@
...
-class account_invoice(osv.Model):
+class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
-    def _fiscal_position_map(self, cr, uid, result, context=None, **kwargs):
+    def _fiscal_position_map(self, result, **kwargs):
+        ctx = dict(self._context)
+        ctx.update({'use_domain': ('use_invoice', '=', True)})
+        return self.env['account.fiscal.position.rule'].with_context(
+            ctx).apply_fiscal_mapping(result, **kwargs)
```
Comment (guewen) [inline account_fiscal_position_rule/account_invoice.py:33]: You don't need to update the context yourself. `with_context()` does that. You can write (with a better care of the length of the line, probably by extracting a part in a variable):

```
return self.env['account.fiscal.position.rule'].with_context(use_domain=('use_invoice', '=', True)).apply_fiscal_mapping(result, **kwargs)
```

**Comment by rvalyi** [comment]
@guewen yes we will put some tests before merging, we just didn't have time so far.

**Comment by max3903** [comment]
:+1:

**Comment by jgrandguillaume** [comment]
:+1:

**Pull request merged**